### PR TITLE
docs: Add "How to contribute to Hugo" tutorial

### PR DESCRIPTION
**Pull request by digitalcraftsman**

### CODE REVIEW

File: docs/content/tutorials/how-to-contribute-to-hugo.md [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+    max-width: 100%;
+    height: auto;
+  } 
+</style>
+
+## Introduction
+
+Hugo is an open source project and lives by the work of its [contributors](https://github.com/spf13/hugo/graphs/contributors). Help to make Hugo even more awesome. There are plenty of [open issues](https://github.com/spf13/hugo/issues) on GitHub and we need your help.
+
+This tutorial intended for people who are new to Git, GitHub or open source projects in general. It should help to overcome most of the barriers that newcomers encounter. It describes step by step what you need to do.
```
Comment (robertbasic) [inline docs/content/tutorials/how-to-contribute-to-hugo.md]: "This tutorial is intended"

File: docs/content/tutorials/how-to-contribute-to-hugo.md [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+
+- Link instructions or describe the setup for Linux/Windows/OSX shortly?
+	- Linux -> apt-get (or installed by default)
+	- Windows -> classical installer
+	- OSX -> classical installer or brew
+- Check installation with `git version`
+
+### Git Graphical Front Ends
+
+There a several [GUI clients](https://git-scm.com/downloads/guis) that help you to operate Git. Not all are available for all operating systems and maybe differ in their usage. Thus, so we will use the command line since the commands are everywhere the same.
```
Comment (robertbasic) [inline docs/content/tutorials/how-to-contribute-to-hugo.md]: There are several

**Comment by digitalcraftsman** [comment]
In the forum and personally I had multiple moments were people asked: "What's the best way to contribute XYZ to Hugo?". This guide took @quoha's [forum thread](https://discuss.gohugo.io/t/hugo-development-how-to-contribute-code/2374) as inspiration. It should help to overcome the barriers that newcomers encounter. 

I would like to hear some feedback from you since I would like finish the sections about the installation of Git and Go.

1. Go can be installed in various ways. Should I just link tutorials that show the neccessary steps or is it better to add a concise description? I personally use GVM because it allows me to switch between different Go versions very easily.
2. The installation of Git is a bit simpler but it's the same as in point one.
3. All shell commands are for UNIX based operating systems. Can we assume that Windows users can research or know the equivalent commands?

File: docs/content/tutorials/how-to-contribute-to-hugo.md [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+```
+
+
+## Set up your working copy
+
+The working copy is set up locally on your computer. It's what you'll edit, compile, and end up pushing back to GitHub. The main steps are cloning the repository and creating your fork as a remote.
+
+### Clone the repository
+
+We assume that you've set up your `GOPATH` (see the section above if you're unsure about this). You should now copy the Hugo repository down to your computer. You'll hear this called "clone the repo". GitHub's [help pages] give us a short explaination:
```
Comment (robertbasic) [inline docs/content/tutorials/how-to-contribute-to-hugo.md]: Help pages are not linked?

Comment (robertbasic) [inline docs/content/tutorials/how-to-contribute-to-hugo.md]: "explanation"

File: docs/content/tutorials/how-to-contribute-to-hugo.md (hunk 2 above)
Comment (digitalcraftsman) [inline docs/content/tutorials/how-to-contribute-to-hugo.md]: Thank you for catching this typos.

File: docs/content/tutorials/how-to-contribute-to-hugo.md [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+### Accept the contributor license agreement
+
+Last but not least you should accept the contributor license agreement (CLA). A new comment should be added automatically to your pull request. Click on the yellow badge, accept the agreement and authenticate yourself with your GitHub account. It just takes a few clicks and only needs to be done once.
+
+<img src="/img/tutorials/how-to-contribute-to-hugo/accept-cla.png" alt="Accept the CLA">
+
+
+### Automatic builds
+
+We use the [Travis CI loop](https://travis-ci.org/spf13/hugo) (Linux and OS&nbsp;X) and [AppVeyor](https://ci.appveyor.com/project/spf13/hugo/branch/master) (Windows) to compile Hugo with your additions. This is should ensure that everything works as expected before merging your pull request. This in most cases only relevant if you made changes to the codebase of Hugo.
```
Comment (robertbasic) [inline docs/content/tutorials/how-to-contribute-to-hugo.md]: "This is should ensure" > "This should ensure"

**Comment by digitalcraftsman** [comment]
Thanks @robertbasic for highlighting some typos.

**Comment by robertbasic** [comment]
@digitalcraftsman you're welcome, happy to help :)

**Comment by JohnBlood** [comment]
@digitalcraftsman , in response to point 3. I could create a quick list of Windows commands that correspond to the UNIX commands. To save Windows users from wasting time.

**Comment by digitalcraftsman** [comment]
@JohnBlood Currently, only `cd` is used. A little Google search revealed it's same command under Windows. This could change if we add installation instructions for Go and Git. See it as reminder.

**Comment by JohnBlood** [comment]
@digitalcraftsman True. about the only other command used is `ls` which is `dir` in Windows.

**Comment by digitalcraftsman** [comment]
I linked installation instructions for Git and Go. Now the guide should be mostly complete.

**Comment by digitalcraftsman** [comment]
If nobody has to complain anything about it I would merge it.

**Comment by digitalcraftsman** [comment]
Merged as ebf3cef1624cbb56e6e6341e2b6f2539a625c370 (master) and f47b4a2de4e28e3b8dbcd15bbecc20cab55c3273 (v0.15.docs).

**Pull request closed without merging**